mavros_extras/src/plugins/distance_sensor.cpp

```diff
@@ -187,6 +188,9 @@ class DistanceSensorPlugin : public plugin::PluginBase {
 		ds.orientation = orientation;
 		ds.covariance = covariance;
 		// [[[end]]] (checksum: 6f6f9449d926ab618a3293e2091c7035)
+		ds.vertical_fov = fov;
+		ds.horizontal_fov = fov;
+		ds.quaternion = {0};
```

Comment:
> It also sets the quaternion to zero (it is not used unless the orientation is set to MAV_SENSOR_ROTATION_CUSTOM, and the message description says: "Set it to 0 if invalid.").

OK so what if `MAV_SENSOR_ROTATION_CUSTOM` is set? It might be reasonable to add an parameter to the range sensor that allows to set it's quaternion instead. Like we have https://github.com/mavlink/mavros/blob/master/mavros/launch/px4_config.yaml#L145.

Reply:
I've added `sensor_orientation` to the set of parameters, which allows the user to specify orientation using roll/pitch/yaw angles. But the `sensor_msgs/Range` message already contains the `frame_id` of the sensor, shouldn't we try and use that instead? The only thing that I'm confused by is the "vehicle body frame": is it FRD? FLU? Is it different from autopilot to autopilot?